addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -53,191 +58,326 @@
-                    if (lastInput != null && !lastInput.equals(previousInput)) {
-                        if (getConfig().get(AUTORUN) != null && ((Boolean) getConfig().get(AUTORUN)).booleanValue()) {
-                            lastInput = command.toString();
+                currentInput = command.toString();
+                if (getConfig().get(RUN_ON_INPUT) != null && ((Boolean) getConfig().get(RUN_ON_INPUT)).booleanValue()) {
+                    if (currentInput != null && currentInput.equals(previousInput)
+                            && getConfig().get(REPEAT_ENABLED) != null
+                            && ((Boolean) getConfig().get(REPEAT_ENABLED)).booleanValue()) {
+                        logger.trace("Executing command '{}' after because of a repitition on the input channel ('{}')",
+                                getConfig().get(COMMAND), command.toString());
+                        scheduler.schedule(new ExecutionRunnable(currentInput), 0, TimeUnit.SECONDS);
+                    } else {
+                        if (currentInput != null && previousInput != null && !currentInput.equals(previousInput)
+                                || (previousInput == null && currentInput != null)) {
                             logger.trace("Executing command '{}' after a change of the input channel to '{}'",
                                     getConfig().get(COMMAND), command.toString());
-                            scheduler.schedule(periodicExecutionRunnable, 0, TimeUnit.SECONDS);
+                            scheduler.schedule(new ExecutionRunnable(currentInput), 0, TimeUnit.SECONDS);
                         }
                     }
                 }
             }
         }
     }
 
-    @Override
-    public void initialize() {
+    class ExecStrLookup extends StrLookup {
 
-        if (executionJob == null || executionJob.isCancelled()) {
-            if (((BigDecimal) getConfig().get(INTERVAL)) != null && ((BigDecimal) getConfig().get(INTERVAL)).intValue() > 0) {
-                int polling_interval = ((BigDecimal) getConfig().get(INTERVAL)).intValue();
-                executionJob = scheduler.scheduleWithFixedDelay(periodicExecutionRunnable, 0, polling_interval,
-                        TimeUnit.SECONDS);
-            }
+        String execInput = "";
+
+        public void setInput(String input) {
+            this.execInput = input;
         }
 
-        updateStatus(ThingStatus.ONLINE);
-    }
+        @Override
+        public String lookup(String key) {
+            try {
+                String transform = null;
+                String format = null;
+                String subkey = "";
+                if (key.contains(":")) {
+                    String parts[] = key.split(":");
+                    subkey = parts[0];
+
+                    if (parts.length == 2) {
+                        format = parts[1];
+                    } else if (parts.length == 3) {
+                        transform = parts[1];
```

Comment:
@martinvw The formatting is done after the transformation, therefore I choose to put it last so that the user would read it that way, e.g. from left to right

Reply:
Okay, if the error message and documentation are clear its fine with me